resources/speechdetector/src/main/java/org/restcomm/media/resource/speechdetector/SpeechDetectorImpl.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2017, Telestax Inc and individual contributors
+ * by the @authors tag. 
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.restcomm.media.resource.speechdetector;
+
+import java.io.IOException;
+
+import org.restcomm.media.component.AbstractSink;
+import org.restcomm.media.spi.memory.Frame;
+
+/**
+ * @author anikiforov
+ *
+ */
+public class SpeechDetectorImpl extends AbstractSink implements SpeechDetector {
```

Comment:
Delete class.

This implementation is too tied to a specific feature, in this case, ASR.
The new speech-detection module should only expose an interface for SpeechDetector. Users should create implementations when they need to, for specific use cases.

Reply:
Another suggestion: SpeechDetectorImpl name is too generic, ideally it should reflect the VAD algorithm that powers it. In the future we will possibly have multiple implementations that follow different strategies/algorithms.